### PR TITLE
Added methods to access real files

### PR DESCRIPTION
**Pull request by mrbean-bremen**
- allow to add really existing files and directory trees to the fake
file system, with the contents read on demand
- see #170

### CODE REVIEW

**Comment by mrbean-bremen** [comment]
@jmcgeheeiv - please review!

File: pyfakefs/fake_filesystem.py [hunk 1]
```diff
@@ -1003,7 +1014,7 @@ def SplitDrive(self, path):
                 # UNC path handling is here since Python 2.7.8, back-ported from Python 3
                 if sys.version_info >= (2, 7, 8):
                     if (path[0:2] == self.path_separator * 2) and (
-                            path[2:3] != self.path_separator):
+                                path[2:3] != self.path_separator):
```
Comment (mrbean-bremen) [inline pyfakefs/fake_filesystem.py]: Didn't notice these changes - PyCharm has sometimes its own ideas about hanging indents, obviously I did an auto-format accidentally. I will revert this.

**Comment by mrbean-bremen** [comment]
I added an example as requested. I had to use `REAL_OPEN` to compare the content with the real file, but you won't need this in a real test, so I think the API can be used in unit tests without problems.

**Comment by mrbean-bremen** [comment]
Also - as `CopyRealFile` has at least one additional feature  (copy to another location), I think it has its own value,  but I don't think we need another addition to TestCase - the methods in fake_filesystem should be sufficient.

**Comment by mrbean-bremen** [comment]
@dwt had similar problems as you mentioned, and after reproducing them (or something related ) I found that the call to `os.path.exists` that is made from the original  `os.listdir`, was the problem in this case. I'm not really sure if this is a good fix for that.

**Comment by mrbean-bremen** [comment]
I may have another go at this later and try to reproduce the problem in a unit test.

**Comment by mrbean-bremen** [comment]
@jmcgeheeiv - I added a test that reproduces the problem. please review!
Note that I rebased against master.

**Comment by mrbean-bremen** [comment]
As to the explanation why the problem happened in the first place: 
Other than the other patched modules, `path` is actually implemented in two modules: `ntpath`/`posixpath` (one of which is imported as `path` by `os`) and `genericpath`, which implements stuff common for both systems. If calling `os.path.exists` from inside the unit test, the real `os` module is used as it should be, but it calls the implementation in `genericpath`, which calls `os.stat` on its part. If `genericpath` is not added to SKIPNAMES, this last `os` will be stubbed out, so `stat` is called on the fake file system.

**Comment by jmcgeheeiv** [comment]
Oh, dear.  Could you resolve this conflict and merge it?

Well done, @mrbean-bremen.

**Comment by mrbean-bremen** [comment]
Done - caused that conflict myself. Thanks!

**Pull request merged**